month_end/models/month_end.py

```diff
@@ -0,0 +1,125 @@
+        gift_debit = 0
+        for gift in montly_gifts:
+            gift_debit += gift.amount
+        gift_line_vals = {
+            'account_id': gift_account.id,
+            'name': gift_name,
+            'debit': gift_debit,
+            'partner_id': partner_compassion_internationnal.id
+        }
+
+        # sponsorship entry
+        sponsorship_account = self.env['account.account'].search([('code', '=', '5000')])
+        sponsorship_active = self.env['recurring.contract'].search_count([('child_id', '!=', False)])
+        sponsorship_disbursed = self.env['recurring.contract'].search_count([('child_id.project_id.hold_cdsp_funds', '=', False)])
+        sponsorship_name = str(sponsorship_active) + ' sponsorships, ' + \
+                           str(sponsorship_disbursed) + ' disbursed'
+
+        sponsorships = self.env['recurring.contract'].search([])
+        sponsorship_debit = 0
+        for sponsorship in sponsorships:
+            for contract in sponsorship.contract_line_ids:
+                sponsorship_debit += contract.amount
+
+        sponsorship_line_vals = {
+            'account_id': sponsorship_account.id,
+            'name': sponsorship_name,
+            'debit': sponsorship_debit,
+            'analytic_account_id': analytic_account_cd_pgm.id,
+            'partner_id': partner_compassion_internationnal.id
+        }
+
+        # USP entry
+        usp_account = self.env['account.account'].search([('code', '=', '5017')])
+        usps = self.env['account.move.line'].search(
+            [
+                ('account_id.code', '=', 5017),
+                ('debit', '>', 0),
+                ('date', '>=', fields.Date.to_string(first_day_last_month)),
+                ('date', '<=', fields.Date.to_string(last_day_last_month))
+            ]
+        )
+        usp_debit = 0
+        for usp in usps:
+            usp_debit += usp.debit
+        usp_line_vals = {
+            'account_id': usp_account.id,
+            'name': 'disbursement',
+            'debit': usp_debit,
+            'analytic_account_id': analytic_account_cd_pgm.id,
+            'partner_id': partner_compassion_internationnal.id
+        }
+
+        # balance entry
+        balance_account = self.env['account.account'].search([('code', '=', '2001')])
+        balance_name = 'disbursement'
+        balance_credit = gift_debit + sponsorship_debit + usp_debit
+        balance_line_vals = {
+            'account_id': balance_account.id,
+            'name': balance_name,
+            'credit': balance_credit,
+            'partner_id': partner_compassion_internationnal.id
+        }
+
+        ## account_move entry
+        journal = self.env['account.journal'].search([('code', '=', 'OD')])[0]
+        ref = today.strftime('%B %Y') + ' disbursement'
+        date = today
+
+        move_vals = {
+            'name': 'Décompte de fin de mois',
```

Comment:
Add `u` prefix to the string to use unicode (good practice when accented characters are in the string)